### PR TITLE
Add dynamic min gas price

### PR DESCRIPTION
**Pull request by karlfloersch**
Add dynamic min gas price logic!

### CODE REVIEW

File: src/batch-submitter/batch-submitter.ts [hunk 1]
```diff
@@ -181,9 +181,16 @@ export abstract class BatchSubmitter {
     this.lastBatchSubmissionTimestamp = Date.now()
     this.log.debug('Waiting for receipt...')
 
+    let minGasPrice = (this.minGasPriceInGwei === 0)
+      ? this.minGasPriceInGwei : ((await this.signer.getGasPrice()).div(1000000000)).toNumber() // convert to gwei
+    if (minGasPrice < this.maxGasPriceInGwei) {
```
Comment (annieke) [inline src/batch-submitter/batch-submitter.ts:186]:
```suggestion
    if (minGasPrice > this.maxGasPriceInGwei) {
```
?

**Review by annieke — COMMENTED** [review]
this makes sense to me! so my change was going to be adding an API call to [gasnow](https://www.gasnow.org/api/v3/gas/price) and not submit if the gas is above the threshold value - is that still necessary? and is there a difference between using gasnow vs `getGasPrice()`?

File: src/batch-submitter/batch-submitter.ts (hunk 1 above)
Comment (karlfloersch) [reply src/batch-submitter/batch-submitter.ts]: Updated in last commit!

**Comment by karlfloersch** [comment]
> this makes sense to me! so my change was going to be adding an API call to [gasnow](https://www.gasnow.org/api/v3/gas/price) and not submit if the gas is above the threshold value - is that still necessary? and is there a difference between using gasnow vs `getGasPrice()`?

So looks like `getGasPrice()` uses `eth_gasPrice` (https://github.com/ethers-io/ethers.js/blob/7f775f7ad634d40bbf2625cde142f7a61e241b8f/packages/providers/src.ts/json-rpc-provider.ts#L399-L400) to return the value. I can definitely update to use gasnow which might be a bit more accurate but one good thing about using `eth_gasPrice` is that I'm confident it works for all testnets

File: src/batch-submitter/batch-submitter.ts (hunk 1 above)
Comment (karlfloersch) [reply src/batch-submitter/batch-submitter.ts]: Also nice catch!!!!

**Review by annieke — APPROVED** [review]
@karlfloersch i think compatibility is important for this one! will ask mark about the threshold change too, might be good to have consistency here

**Review by tynes — APPROVED** [review]

**Pull request merged**